### PR TITLE
CMC copies Pitzer's backend + slow but potential Mudd backend fix

### PR DESCRIPTION
**Pull request by mfeng1904**
load_menu on vagrant: http://puu.sh/rW31q/777744ab87.png
localhost: http://puu.sh/rW32E/3855480cd5.png

CMC's backend is now the same as Pitzer's. The two schools' site menu layout is identical, but the reason CMC broke last week was because its backend does a i in range 7 loop, which doesn't fly when the dining hall isn't open for all 7 days of the week. Meanwhile, Pitzer's backend handles shorter weeks, so I just did a copy paste...

Since Mudd changed their site menu layout, I wrote a new backend using Selenium. On the Mudd site, each day is on a new tab, so I ask Selenium to click through each day in order to scrape the entire week's menu. As a result, the current Mudd backend is much slower than the others, but gets the job done presuming the website doesn't time out on it... Furthermore, Mudd's menu doesn't display on localhost, but since the menu model's values + food item lists all look right, it should display unless something went awry in menu.food_item.add() or .save(). Additionally, CMC's menu isn't displaying on localhost at the moment and if Pitzer is working, CMC should work too.

Hopefully both these menus will display during production. Worst case I imagine Mudd will just stay broken, and I can easily revert CMC

### CODE REVIEW

File: aspc/menu/backends/ghostdriver.log [hunk 1]
```diff
@@ -0,0 +1,215 @@
+[INFO  - 2016-10-23T08:46:18.937Z] GhostDriver - Main - running on port 53800
```
Comment (KentShikama) [inline aspc/menu/backends/ghostdriver.log]: Remove this log file from git

File: aspc/menu/backends/mudd.py [hunk 2]
```diff
@@ -1,65 +1,118 @@
...
+        """
+        try:
+            self.selenium.get(self.menu_url)
+        except:
+            print "Error: Mudd's website likely timed out or sent a bad response. Returning days that were successfully scraped."
+            return self.menus
+        
+        #toggle through each day's button and scrape the updated menu.
+        #skip if button not found (in case of shorter week)
+        for i in range(7):
```
Comment (KentShikama) [inline aspc/menu/backends/mudd.py]: Extract this into a method

See http://refactoring.com/catalog/extractMethod.html

File: ghostdriver.log [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[INFO  - 2016-10-26T08:53:58.878Z] GhostDriver - Main - running on port 45320
```
Comment (KentShikama) [inline ghostdriver.log]: Another log file

File: aspc/settings.py.example [hunk 4]
```diff
@@ -95,11 +95,10 @@ FACEBOOK_APP_SECRET = ''
...
 BING_SEARCH_KEY = ''
-
 GOOGLE_SEARCH_KEY = ''
 GOOGLE_SEARCH_CX = ''
 AMAZON_ACCESS_KEY = ''
 AMAZON_SECRET_KEY= ''
 AMAZON_ASSOC_TAG = ''
 
-GEOPOSITION_GOOGLE_MAPS_API_KEY = ''
+GEOPOSITION_GOOGLE_MAPS_API_KEY = ''
```
Comment (KentShikama) [inline aspc/settings.py.example]: @ZiqiXiong Did you not already commit this?

File: vagrant/init/init_as_user.sh [hunk 5]
```diff
@@ -45,28 +45,22 @@ then
     cp /vagrant/aspc/settings.py.example /vagrant/aspc/settings.py
 fi
 
-# create tables
```
Comment (KentShikama) [inline vagrant/init/init_as_user.sh]: Don't delete the migrations or collectstatic

File: aspc/menu/management/commands/load_menus.py [hunk 6]
```diff
@@ -8,6 +8,7 @@
 import json
 import traceback
 import ast
+import ipdb
```
Comment (KentShikama) [inline aspc/menu/management/commands/load_menus.py]: Remove this; should be only for personal debugging

File: aspc/menu/backends/mudd.py [hunk 7]
```diff
@@ -1,65 +1,118 @@
...
+                    if menu_button == "changeTab(%d)"%(i):
+                        day_name = button.get_attribute("innerHTML")
+                        button.click()
+                        self.menus[day_name] = self.get_day_menu()
+                        print "Scraped Mudd-" + day_name
+                        break
+            except:
+                print "Error: Mudd's website likely timed out or sent a bad response. Returning days that were successfully scraped."
+                return self.menus
+        #Pretty(ish) print for debugging
```
Comment (KentShikama) [inline aspc/menu/backends/mudd.py]: Another method extraction or simply delete?

File: aspc/menu/backends/mudd.py [hunk 8]
```diff
@@ -1,65 +1,118 @@
...
+        }
+        try:
+            self.menu_url = self.get_menu_url()
+        except:
+            print "Error: Menu link currently unavailable from website"
+            return
+    
+    def get_menu_url(self):
+        self.selenium.get(self.homepage_url)
+        link_container = self.selenium.find_element_by_id("ui-accordion-accordion_3543-panel-0")
```
Comment (KentShikama) [inline aspc/menu/backends/mudd.py]: This looks so fragile. Why is it 3543? Will it change to 3544 sometime? Is there no other way to get the menu url?

File: aspc/settings.py.example (hunk 4 above)
Comment (ZiqiXiong) [inline aspc/settings.py.example]: yes

File: aspc/menu/backends/mudd.py (hunk 8 above)
Comment (KentShikama) [inline aspc/menu/backends/mudd.py]: Yeah `menu_url = doc.find_all('div', {'class': 'accordionBody'})[0].find_all('a')[0]['href']` looks a bit better. I'll have to check the performance of this.

File: aspc/settings.py.example (hunk 4 above)
Comment (KentShikama) [inline aspc/settings.py.example]: @ZiqiXiong So what is going on lol

Comment (KentShikama) [inline aspc/settings.py.example]: @ZiqiXiong Oh nvm it is just a reformatting

**Pull request merged**